src/test/rpc/AccountTx_test.cpp

```diff
@@ -120,22 +120,37 @@ class AccountTx_test : public beast::unit_test::suite
         // All other ledgers have no txs
 
         auto hasTxs = [apiVersion](Json::Value const& j) {
-            return j.isMember(jss::result) &&
-                (j[jss::result][jss::status] == "success") &&
-                (j[jss::result][jss::transactions].size() == 2) &&
-                (j[jss::result][jss::transactions][0u][jss::tx]
-                  [jss::TransactionType] == jss::AccountSet) &&
-                (j[jss::result][jss::transactions][1u][jss::tx]
-                  [jss::TransactionType] == jss::Payment) &&
-                (j[jss::result][jss::transactions][1u][jss::tx]
-                  [jss::DeliverMax] == "10000000010") &&
-                ((apiVersion > 1 &&
-                  !j[jss::result][jss::transactions][1u][jss::tx].isMember(
-                      jss::Amount)) ||
-                 (apiVersion <= 1 &&
-                  j[jss::result][jss::transactions][1u][jss::tx][jss::Amount] ==
-                      j[jss::result][jss::transactions][1u][jss::tx]
-                       [jss::DeliverMax]));
+            switch (apiVersion)
+            {
+                case 1:
+                    return j.isMember(jss::result) &&
+                        (j[jss::result][jss::status] == "success") &&
+                        (j[jss::result][jss::transactions].size() == 2) &&
+                        (j[jss::result][jss::transactions][0u][jss::tx]
+                          [jss::TransactionType] == jss::AccountSet) &&
+                        (j[jss::result][jss::transactions][1u][jss::tx]
+                          [jss::TransactionType] == jss::Payment) &&
+                        (j[jss::result][jss::transactions][1u][jss::tx]
+                          [jss::DeliverMax] == "10000000010") &&
+                        (j[jss::result][jss::transactions][1u][jss::tx]
+                          [jss::Amount] ==
+                         j[jss::result][jss::transactions][1u][jss::tx]
+                          [jss::DeliverMax]);
+                case 2:
+                    return j.isMember(jss::result) &&
+                        (j[jss::result][jss::status] == "success") &&
+                        (j[jss::result][jss::transactions].size() == 2) &&
+                        (j[jss::result][jss::transactions][0u][jss::tx_json]
+                          [jss::TransactionType] == jss::AccountSet) &&
+                        (j[jss::result][jss::transactions][1u][jss::tx_json]
+                          [jss::TransactionType] == jss::Payment) &&
+                        (j[jss::result][jss::transactions][1u][jss::tx_json]
+                          [jss::DeliverMax] == "10000000010") &&
+                        (!j[jss::result][jss::transactions][1u][jss::tx_json]
+                              .isMember(jss::Amount));
```

Comment:
To be thorough, we can add the checks for other fields like `jss::hash`, `jss::close_time_iso` etc

Reply:
Great idea, and actually helped me uncover a bug (see last commit)

Reply:
hmm, I tested it myself but I didn't run into any bugs. anyway, I'm glad you found it 👍

Reply:
oh, you had called `txnMeta->getIndex` instead of `txn->getLedger` :)
Honestly, I find it hard to tell the differences between `TxMeta::mIndex`, `TxMeta::mTransactionID`, `TxMeta::mLedger` --> they have similar types and appear to fulfill similar functionality.

Reply:
Yeah, you are absolutely right :cry: Luckily there's a simple strategy to avoid bugs this causes i.e. strong typing - but that's refactoring work, definitely not this PR

Reply:
yeah! Strong typing is a great idea